packages/pipelines/src/utils/tekton-results-utils.ts

```diff
@@ -0,0 +1,455 @@
+              : '';
+          case 'LessThan':
+            return expression.values?.[0]
+              ? EXP(`data.metadata.labels["${expression.key}"]`, expression.values?.[0], '<')
+              : '';
+          default:
+            throw new Error(
+              `Tekton results operator '${expression.operator}' conversion not implemented.`,
+            );
+        }
+      })
+      .filter((x) => x),
+  );
+
+export const selectorToFilter = (selector: Selector) => {
+  let filter = '';
+  if (selector) {
+    const { matchLabels, matchExpressions, filterByName } = selector;
+
+    if (filterByName) {
+      filter = AND(filter, nameFilter(filterByName as string));
+    }
+
+    if (matchLabels || matchExpressions) {
+      if (matchLabels) {
+        filter = AND(filter, labelsToFilter(matchLabels));
+      }
+      if (matchExpressions) {
+        filter = AND(filter, expressionsToFilter(matchExpressions));
+      }
+    } else {
+      filter = labelsToFilter(selector as MatchLabels);
+    }
+  }
+  return filter;
+};
+
+// Devs should be careful to not cache a response that may not be complete.
+// In most situtations, caching is unnecessary.
+// Only cache a response that returns a single complete record as lists can change over time.
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+let CACHE: { [key: string]: [any[], RecordsList] } = {};
+export const clearCache = () => {
+  CACHE = {};
+};
+const InFlightStore: { [key: string]: boolean } = {};
+let cachedTektonResultsAPI: string;
+
+export const getTektonResultsAPIUrl = async (fetchUtils?: FetchUtilsType) => {
+  if (!fetchUtils?.hooks?.k8sGet) {
+    return cachedTektonResultsAPI;
+  }
+  if (cachedTektonResultsAPI) {
+    return cachedTektonResultsAPI;
+  } else {
+    const cachedTektonResult: K8sResourceKind = await fetchUtils?.hooks?.k8sGet({
+      model: TektonResultModel,
+      name: 'result',
+    });
+    const targetNamespace = cachedTektonResult?.spec?.targetNamespace;
+    const serverPort = cachedTektonResult?.spec?.server_port ?? '8080';
+    const tlsHostname = cachedTektonResult?.spec?.tls_hostname_override;
+    if (tlsHostname) {
+      cachedTektonResultsAPI = `${tlsHostname}:${serverPort}`;
+    } else if (targetNamespace && serverPort) {
+      cachedTektonResultsAPI = `tekton-results-api-service.${targetNamespace}.svc.cluster.local:${serverPort}`;
+    } else {
+      cachedTektonResultsAPI = `tekton-results-api-service.openshift-pipelines.svc.cluster.local:${serverPort}`;
+    }
+    return cachedTektonResultsAPI;
+  }
+};
+
+export const createTektonResultsUrl = async (
+  namespace: string,
+  dataType: DataType,
+  filter?: string,
+  options?: TektonResultsOptions,
+  nextPageToken?: string,
+  tektonResultsBaseURL?: string,
+  fetchUtils?: FetchUtilsType,
+): Promise<string> => {
+  const tektonResultsAPI = tektonResultsBaseURL
+    ? tektonResultsBaseURL
+    : await getTektonResultsAPIUrl(fetchUtils);
```

Comment:
Based on my previous comment, we should not be having any logic to figure out the URL.